modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexTaskStateUpdater.java

```diff
@@ -234,15 +235,24 @@ public void onFailure(Exception e) {
                             TimeValue nextDelay = getNextDelay(delay);
                             logger.info(new ParameterizedMessage("Failed to read from {} index on FINISHED, retrying in {}",
                                 REINDEX_INDEX, nextDelay), e);
-                            threadPool.schedule(() -> writeFinishedState(reindexResponse, exception, nextDelay), nextDelay,
-                                ThreadPool.Names.SAME);
+                            reschedule(nextDelay, reindexResponse, exception);
                         }
                     });
                 } else {
                     TimeValue nextDelay = getNextDelay(delay);
                     logger.info(new ParameterizedMessage("Failed to write to {} index on FINISHED, retrying in {} [task-id={}]",
                         REINDEX_INDEX, nextDelay, taskId), e);
+                    reschedule(nextDelay, reindexResponse, exception);
+                }
+            }
+
+            private void reschedule(TimeValue nextDelay, @Nullable BulkByScrollResponse reindexResponse,
+                                    @Nullable ElasticsearchException exception) {
+                try {
                     threadPool.schedule(() -> writeFinishedState(reindexResponse, exception, nextDelay), nextDelay, ThreadPool.Names.SAME);
+                } catch (EsRejectedExecutionException rejection) {
```

Comment:
Use scheduleUnlessShuttingDown?

Reply:
Thanks, fixed in ae6e8b2e527fa30d2f6a8f2aa7840c16d5cd304a